Update cognito_user_pool.markdown

ehault opened the pull request:
Fix documentation error on the verify by link for Cognito Pool.

With {##VERIFY EMAIL##} we get:
 
```
* module.cognito.aws_cognito_user_pool.pool: 1 error(s) occurred:

* aws_cognito_user_pool.pool: Error updating Cognito User pool: InvalidParameterException: The message is invalid.
        status code: 400, request id: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxx
```

But with this change ( {##Click Here##} ) we get the proper response:

```
Apply complete! Resources: 0 added, 1 changed, 0 destroyed.
```

bflad commented:
Hi @ehault 👋 Could you provide some additional context about your configuration? I just verified that `{##VERIFY EMAIL##}` successfully wrote to Cognito by updating this line:

https://github.com/terraform-providers/terraform-provider-aws/blob/d095a2c72b372b405fdc1ed337024c8f7dc4bc71/aws/resource_aws_cognito_user_pool_test.go#L1126

And running the acceptance test (without updating the test condition, so it fails as expected):

```
--- FAIL: TestAccAWSCognitoUserPool_withVerificationMessageTemplate (9.12s)
    testing.go:538: Step 0 error: Check failed: 1 error occurred:
        	* Check 4/7 error: aws_cognito_user_pool.pool: Attribute 'verification_message_template.0.email_message_by_link' expected "{##foobar##}", got "{##VERIFY EMAIL##}"
```

Is your message almost 20,000 characters long by chance? The [API reference for `VerificationMessageTemplateType`](https://docs.aws.amazon.com/cognito-user-identity-pools/latest/APIReference/API_VerificationMessageTemplateType.html) notes these restrictions (which we do not currently validate during plan time), however the content within `{##...##}` shouldn't matter either which way between the old and new configuration you provided.

> **EmailMessageByLink**
>
> The email message template for sending a confirmation link to the user.
> 
> Type: String
> 
> Length Constraints: Minimum length of 6. Maximum length of 20000.
> 
> Pattern: [\p{L}\p{M}\p{S}\p{N}\p{P}\s*]*\{##[\p{L}\p{M}\p{S}\p{N}\p{P}\s*]*##\}[\p{L}\p{M}\p{S}\p{N}\p{P}\s*]*
> 
> Required: No

ehault commented:
I updated my pool to go from "CONFIRM_WITH_CODE" to "CONFIRM_WITH_LINK",
and updated the email to have {##VERIFY EMAIL##} and that failed with the
above error message. Then after some google-fu I found an old post saying
to use the {##CLICK HERE##} approach, which then worked.

Not sure what that could be about though...

Thanks for your help, do not hesitate to drop this altogether.

Sincerely

Thibault LEPEU


On Wed, Oct 31, 2018 at 1:38 PM Brian Flad <notifications@github.com> wrote:

> Hi @ehault <https://github.com/ehault> 👋 Could you provide some
> additional context about your configuration? I just verified that {##VERIFY
> EMAIL##} successfully wrote to Cognito by updating this line:
>
>
> https://github.com/terraform-providers/terraform-provider-aws/blob/d095a2c72b372b405fdc1ed337024c8f7dc4bc71/aws/resource_aws_cognito_user_pool_test.go#L1126
>
> And running the acceptance test (without updating the test condition, so
> it fails as expected):
>
> --- FAIL: TestAccAWSCognitoUserPool_withVerificationMessageTemplate (9.12s)
>     testing.go:538: Step 0 error: Check failed: 1 error occurred:
>         	* Check 4/7 error: aws_cognito_user_pool.pool: Attribute 'verification_message_template.0.email_message_by_link' expected "{##foobar##}", got "{##VERIFY EMAIL##}"
>
> Is your message almost 20,000 characters long by chance? The API
> reference for VerificationMessageTemplateType
> <https://docs.aws.amazon.com/cognito-user-identity-pools/latest/APIReference/API_VerificationMessageTemplateType.html>
> notes these restrictions (which we do not currently validate during plan
> time), however the content within {##...##} shouldn't matter either which
> way between the old and new configuration you provided.
>
> *EmailMessageByLink*
>
> The email message template for sending a confirmation link to the user.
>
> Type: String
>
> Length Constraints: Minimum length of 6. Maximum length of 20000.
>
> Pattern: [\p{L}\p{M}\p{S}\p{N}\p{P}\s*]*{##[\p{L}\p{M}\p{S}\p{N}\p{P}\s*]
> *##}[\p{L}\p{M}\p{S}\p{N}\p{P}\s*]*
>
> Required: No
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/terraform-providers/terraform-provider-aws/pull/6309#issuecomment-434689880>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABAdtft7wB5xD9J57aBaUHF8EN7r2Fakks5uqae6gaJpZM4YD5XO>
> .
>
